pkg/proctree/datasource.go

```diff
@@ -0,0 +1,210 @@
+package proctree
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/aquasecurity/tracee/types/datasource"
+	"github.com/aquasecurity/tracee/types/detect"
+)
+
+// DataSource is an implementation to detect.Datasource interface, enveloping the ProcessTree type.
+// It exposes all the relevant information of the tree using the interface methods.
+type DataSource struct {
+	procTree *ProcessTree
+}
+
+func NewDataSource(processTree *ProcessTree) *DataSource {
+	return &DataSource{procTree: processTree}
+}
+
+func (ptds *DataSource) Get(key interface{}) (map[string]interface{}, error) {
+	switch typedKey := key.(type) {
+	case datasource.ProcKey:
+		process, found := ptds.procTree.GetProcessByHash(typedKey.Hash)
```

Comment:
Yea, because the PID is insufficient to be a unique identifier.
In the future we can add a method of caching PIDs` hashes, but it is a feature and doesn't require to change current API.